Update and Improve install.md (#3158)
ILIAS-eLearning/ILIAS: docs/configuration/install.md

# ILIAS Installation

> Around since 1998, ILIAS is a powerful learning management system that fulfils

Reviewer: ```suggestion
Since around 1998, ILIAS is a powerful learning management system that fulfils
```

Author: I'll simply remove this, probably does not belong in installation instructions anyway.

The necessary hardware to run an ILIAS installation is always dependent from the number of users and the kind of usage.

<a name="hardware"></a>
## Hardware

> We recommend a **common dual core server CPU**. Memory requirements are greatly

Reviewer: Do we really want this block? Personally I would never run ILIAS on a setup like this unless I'm running a very small instance with all goodies switched off. Our instance is not huge, but I'm still counting around 50 Cores, multiple Terabytes of data (without counting RAIDs and failovers) and 100's of MB of RAM. Additionally running ILIAS on a hard drive will be painful and clearly not comply with today's demands for snappiness. 100 Mbits does also sound slightly 1990's ;-).

Just a suggestion from where I stand we can also simply throw it out:
"The hardware requirements for ILIAS vary widely, depending on the number of concurrent users you expect and the features you want to enable. Please be aware that ILIAS is not a webpage, but a highly interactive application, thus requirements will be higher than in case of the former. Snappiness of the system will highly depend on deploying enough resources and tailoring the system to your needs. In any case we recommend an absolute minimum of a **common dual core server CPU**, **4GB of RAM**, and a **100 Mbit/s** internet connection. Roughly estimated the operating system and ILIAS itself will use around 25GB. From there you can calculate your storage needs based on the amount of files and media content you expect to upload plus a few GBs for the database.

On MySQL 5.8+ and Galera the `Strict SQL Mode` must be disabled. See [MySQL Strict Mode](#mysql-strict-mode-56) for details.

On MySQL/MariaDB `innodb_large_prefix` must be set to `OFF` if the `ROW_FORMAT`
is set to `COMPACT`.

<a name="reference-system"></a>
## Reference System

The ILIAS Testserver (https://test7.ilias.de) is currently configured as follows:

| Package        | Version                     |
|----------------|-----------------------------|
| Distribution   | Ubuntu 20.04 LTS            |
| MariaDB        | 10.0.38                     |
| PHP            | 7.3                         |
| Apache2        | 2.4.18                      |
| zip            | 3.0                         |
| unzip          | 6.00                        |
| JDK            | OpenJDK 8                   |
| Node.js        | 10.23.0                     |
| wkhtmltopdf    | 0.12.5                      |
| Ghostscript    | 9.26                        |
| Imagemagick    | 6.8.9-9 Q16                 |

<a name="other-platforms"></a>
## Other Platforms or Configurations

Please note that other platforms and configurations should be possible, but it

> might be harder to find someone who can help when things go south. You shall not

Reviewer: Sounds a little bit like Gandalf :-)
```suggestion
might be harder to find someone who can help when things go south. You should not
```

systemctl restart httpd.service
```

<a name="php-installation"></a>
### PHP Installation and Configuration

On Debian/Ubuntu execute:
```

> apt-get install libapache2-mod-php7.3 php7.3-gd php7.3-mysql php7.3-mbstring php7.3-curl php7.3-dom php7.3-zip php7.3-xml

Reviewer: I would recommend to get rid of this. It is hard to keep track of and changes can be sudden. Additionally, if you need this instructions and follow them by the letter, you may end up with a system including eol packages really quickly.

My suggestion:
```suggestion
Please refere to the documentation of your installation to install either php7.3 or php7.4 including the gd, mysql, mbstring, curl, dom, zip, and xml modules. 
```

```

On Debian/Ubuntu execute:
```
apt-get install zip unzip imagemagick openjdk-8-jdk
```

On RHEL/CentOS execute:
```
yum install zip unzip libxslt ImageMagick java-1.8.0-openjdk
```

Restart the apache webserver after you installed dependencies!

<a name="optional-dependencies"></a>
### Optional Dependencies

Depending on your use case, you MAY want to install further dependencies (exact package names vary by distribution and PHP version you are using):

* php7.3-curl
* php7.3-xmlrpc
* php7.3-soap
* php7.3-ldap
* ffmpeg
* mimetex

Restart the apache webserver after you installed dependencies!

<a name="get-code"></a>
## Get the Code

You can [download](http://www.ilias.de/docu/goto.php?target=st_229) the latest
ILIAS release or clone it from [GitHub](https://github.com/ILIAS-eLearning/ILIAS).
For production use make sure to checkout the latest stable release, not the trunk,
which is the development branch of the repository.

We recommend to clone from GitHub and use git to update the code, since this simplifies
the update to future releases and versions.

Clone the code to the web servers docroot (e.g. `/var/www/html`) with the following
commands:

```
cd /var/www/html/

> git clone https://github.com/ILIAS-eLearning/ILIAS.git .

Reviewer: We could suggest to use --single-branch here. For me the difference when testing was 500MB of dead waste.

git checkout release_7
```

or unpack the downloaded archieve to the docroot

The repository of ILIAS doesn't contain all code that is required to run. To
download the required PHP-dependencies and to create static artifacts from
the source, run the following in your ILIAS folder:

```
composer install --no-dev
```

This requires that the php dependency manager [composer](https://getcomposer.org/)
is available in your $PATH.

Create a directory outside the web servers docroot (e.g. `/var/www/files`). Make
sure that the web server is the owner of the files and directories that were created
by changing the group and owner to www-data (on Debian/Ubuntu) or apache (on RHEL).

```
chown www-data:www-data `/var/www/html
chown www-data:www-data `/var/www/files
```

> If you want to server other applications from your webserver, you might want to

Reviewer: ```suggestion
If you want to serve other applications from your webserver, you might want to
```
...but I would actually suggest to kick that part out and only keep "The commands above will directly serve ILIAS from the docroot." I don't see many good reasons to want to serve ILIAS from subdirectories. Certificates are free (or at least getting cheaper if you want e.Vs.) and subdomains seem by far more elegant to me.

to a location outside your docroot. Fill in the configuration fields that are already
contained in the minimal config. Have a look into the [list of available config options](../../setup/README.md#about-the-config-file)
and add the fields that your environment and installation requires. A typical
configuration might look like this afterwards:

```
{
	"common" : {
		"client_id" : "myilias"
	},
	"database" : {
		"user" : "ilias_user",
        "password" : "my_password"
	},
	"filesystem" : {
		"data_dir" : "/var/www/files"
	},
	"http" : {
		"path" : "http://demo1.cat06.de"
	},
    "language" : {
        "default_language" : "de",
        "install_languages" : ["de"]
    },
    "loggin" : {
        "enable" : true,
        "path_to_logfile" : "/var/www/logs/ilias.log",
        "errorlog_dir" : "/var/www/logs/"
    },
	"systemfolder" : {
		"contact" : {
			"firstname" : "Richard",
			"lastname" : "Klees",
			"email" : "richard.klees@concepts-and-training.de"
		}
	},
    "utilities" : {
        "path_to_convert" : "/usr/bin/convert",
        "path_to_zip" : "/usr/bin/zip",
        "path_to_zip" : "/usr/bin/unzip"
    }
}
```

> Run the ILIAS command line setup with your configuration file from your ILIAS folder:

Reviewer: ```suggestion
Run the ILIAS command line setup from within your ILIAS folder passing your configuration file as a parameter:
```

```

> php setup/setup.php install my-configuration.json

Reviewer: ```suggestion
php setup/setup.php install /foo/bar/my-configuration.json
```

## Configure WebDAV (optional)

The recommended webserver configuration is either **Apache with mod_php** or
**Nginx with PHP-FPM (> 1.3.8)**. Do NOT use **Apache with PHP-FPM** if you
use WebDAV.

<a name="webdav-windows-explorer"></a>
### WebDAV with Windows Explorer
Because of a special behaviour in the Windows Explorer, it sometimes fails to
add a WebDAV connection with the error code `0x80070043 The Network Name
Cannot Be Found`.

To prevent this behaviour, add the following rewrite rules to a .htaccess file
in your webroot or to the corresponding section of the configuration of your
webserver:

```
RewriteCond %{HTTP_USER_AGENT} ^(DavClnt)$
RewriteCond %{REQUEST_METHOD} ^(OPTIONS)$
RewriteRule .* "-" [R=401,L]
```

<a name="webdav-mac-finder"></a>
### WebDAV with Mac Finder
To upload files, the WebDAV Client *Finder* on Mac uses chunked transfer encoding.
Some webservers can't handle this way of uploading files and are serving ILIAS
an empty files, which results in an empty file object on ILIAS. Due to a bug in
apache, the configuration of **Apache with PHP-FPM** does not work with the *Mac
Finder*. If you use WebDAV on your ILIAS installation, we recommend to either

> use **Apache with mod_php** or **Nginx with PHP-FPM (> 1.3.8)**.

Reviewer: ```suggestion
use **Apache with mod_php** or **Nginx with PHP-FPM (> 1.3.8)**. Please find more information in the corresponding [README](../../Services/WebDAV/README.md)
```
...and then we get rid of the rest of the information in here concerning WebDAV. We might want to do this for all optional elements to keep this file short and hand responsibility over to maintainers. What do you think?

Author: Something like this would be great.
